src/checker.ml

```diff
@@ -877,7 +900,17 @@ let entrypoint_receive_price (state, price: checker * Ligo.nat) : (LigoOp.operat
   if !Ligo.Tezos.sender <> state.external_contracts.oracle then
     (Ligo.failwith error_UnauthorisedCaller : LigoOp.operation list * checker)
   else
-    (([]: LigoOp.operation list), {state with last_price = Some price})
+    (([]: LigoOp.operation list), {state with last_index = Some price})
+
+let entrypoint_receive_ctez_marginal_price (state, _price: checker * (Ligo.nat * Ligo.nat)) : (LigoOp.operation list * checker) =
+  assert_checker_invariants state;
+  if !Ligo.Tezos.sender <> state.external_contracts.ctez_cfmm then
+    (Ligo.failwith error_UnauthorisedCaller : LigoOp.operation list * checker)
+  else
+    (* FIXME: Figure out how to interpret the received fraction instead of
+     * hardwiring it to one_ratio. *)
+    (([]: LigoOp.operation list), {state with last_ctez_in_tez = Some one_ratio})
```

Comment:
@dorranh instead of opening a new issue I opted for leaving #239 open for a little longer (see https://github.com/tezos-checker/checker/issues/239#issuecomment-956032279).